src/components/pages/ImageClassification/ImageAnnotationModal/ImageAnnotationModal.js

```diff
@@ -97,12 +109,25 @@ const ImageAnnotationModal = ({
 
           setDataToReview({ ...data, points: formattedPoints })
         })
-        .catch(() => {
-          toast.error('Failed to fetch image annotations')
+        .catch((error) => {
+          handleHttpResponseError({
+            error,
+            callback: () => {
+              toast.error(
+                ...getToastArguments(`Failed to fetch image annotations. ${error.message}`),
+              )
+            },
+            shouldShowServerNonResponseMessage: false,
+          })
         })
     }
-    // eslint-disable-next-line
-  }, [])
```

Comment:
There are a lot of these for IC. Im trying to remove ignoring exhaustive deps errors in linting when easy enough to prevent bugs. In this case all it took was using some `useCallback`s